test/ConfigTestCases.test.js

```diff
@@ -277,6 +277,9 @@ describe("ConfigTestCases", () => {
 													: fit(title, () => {}).pend("Skipped")
 										);
 									});
+									// workaround for jest running clearSpies on the wrong suite (invoked by clearResourcesForRunnable)
```

Comment:
It's a bug in `jest` that is triggered by generating tests dynamically. But its possible it could also be triggered in another context.

https://github.com/facebook/jest/blob/440e9c6012bb5079c673a2b7aa8e2a821e99242b/packages/jest-jasmine2/src/jasmine/Env.js#L100-L103

1. In the context of `clearResourcesForRunnable` we have the `id` of the suite or test that should be cleared.
2. Inside, we call `spyRegistry.clearSpies` without passing that id. 
3. `clearSpies` then tries to figure out that ID by calling `currentSpies()`
4. The runnable ID is resolved from `currentRunnable().id`.
5. `currentRunnable` goes to call `currentSuite()`
6. which simply returns the Suite that was started most recently (whenever a Suite is started, we stash it into `currentlyExecutingSuites` Array)

As you can see, the logic is quite convoluted.

I'm still not entirely sure why the `Suite` stack order is incorrect, but setting `suite.disable` makes `jest` skip the garbage collection on it, which solves the issue. The suites run even though they're marked as `disabled`, since we're passing them directly into `execute`.